x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/job/process/autodetect/state/ModelState.java

```diff
@@ -43,9 +43,13 @@ public static String extractJobId(String docId) {
      * and ended with hash and an integer.
      */
     private static String v54ExtractJobId(String docId) {
-        Matcher matcher = V_5_4_DOC_ID_REGEX.matcher(docId);
+        int potentialSuffixIndex = docId.lastIndexOf('-');
+        if (potentialSuffixIndex <= 0 || potentialSuffixIndex >= docId.length() - 1) {
```

Comment:
It's true that this part is unnecessary, but it doesn't hurt, and the CI is green so I'll leave it in rather than do another iteration just for this. It avoids creating an object unnecessarily if it's obviously not going to match, so it's not all bad.